lib/mongo_compat_api.js

```diff
@@ -0,0 +1,325 @@
+var Promise = require("bluebird");
+var assert = require("assert");
+
+/**
+ * Implements a MongoDB Node.js 2.0 driver compatible API that works for
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (ditcher, params) {
```

Comment:
do we really need to pass `ditcher` here? would it better if we just pass the db configuration here? This module itself can then create the `ditcher` instance for shared app dbs, but then use mongodb client directly for apps with standalone dbs?